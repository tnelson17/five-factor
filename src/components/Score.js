import React from 'react';

const Score = (props) => {
  return (
    <div>{props.score}</div>
  )
}

export default Score;
