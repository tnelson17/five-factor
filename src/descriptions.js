const descriptions = {
  O: {
    low: "ur not open lmao",
    moderateLow: "ur not open lmao",
    moderate: "ur kinda open lmao",
    moderateHigh: "ur kinda open lmao",
    high: "ur totally open bro!"
  },
  C: {
    low: "ur not conscientious lmao",
    moderateLow: "ur not conscientious lmao",
    moderate: "ur kinda conscientious lmao",
    moderateHigh: "ur kinda conscientious lmao",
    high: "ur totally conscientious bro!"
  },
  E: {
    low: "ur not extroverted lmao",
    moderateLow: "ur not extroverted lmao",
    moderate: "ur kinda extroverted lmao",
    moderateHigh: "ur kinda extroverted lmao",
    high: "ur totally extroverted bro!"
  },
  A: {
    low: "ur not agreeable lmao",
    moderateLow: "ur not agreeable lmao",
    moderate: "ur kinda agreeable lmao",
    moderateHigh: "ur kinda agreeable lmao",
    high: "ur totally agreeable bro!"
  },
  N: {
    low: "ur not neurotic lmao",
    moderateLow: "ur not neurotic lmao",
    moderate: "ur kinda neurotic lmao",
    moderateHigh: "ur kinda neurotic lmao",
    high: "ur totally neurotic bro!"
  }
}

export default descriptions;
