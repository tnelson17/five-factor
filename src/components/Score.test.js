import React from 'react';
import Score from './Score';
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme';
import descriptions from '../descriptions.js'

it('renders without crashing', () => {
  shallow(
    <Score
      score={0}
      domainKey={'O'}
      domain={"Openness to Experience"}
      max={24}
    />
  )
})

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Score
        score={0}
        domainKey={'O'}
        domain={"Openness to Experience"}
        max={24}
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
