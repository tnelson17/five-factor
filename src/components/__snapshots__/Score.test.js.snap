// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div>
    Your score in the dimension of 
    Openness to Experience
     is 
    0
     out of 
    24
  </div>,
  <div>
    You are very rarely playful, curious, imaginative, creative,
    open-minded, novelty-seeking, forward-looking
  </div>,
]
`;
